### PR TITLE
Use the github repository associated with the PullRequest object to p…

### PR DESCRIPTION
**Pull request by cpennington**
…ost a message

### CODE REVIEW

File: tubular/github_api.py [hunk 1]
```diff
@@ -729,22 +730,23 @@ def _not_duplicate(pr_messages, new_message):
...
+            except UnknownObjectException:
+                raise InvalidPullRequestError('PR #{pr_number} does not exist'.format(pr_number=pr_number))
 
         if force_message or _not_duplicate(pull_request.get_issue_comments(), message_filter):
             return pull_request.create_issue_comment(message)
         else:
             return None
 
-    def message_pr_deployed_stage(self, pr_number, deploy_date=None, force_message=False):
+    def message_pr_deployed_stage(self, pr, deploy_date=None, force_message=False):
```
Comment (macdiesel) [inline tubular/github_api.py]: pull_request is used above we should pick one and make the naming consistent.

**Review by macdiesel — APPROVED** [review]
A nit but makes sense otherwise.

File: tubular/github_api.py (hunk 1 above)
Comment (cpennington) [reply tubular/github_api.py]: Makes sense.

File: tubular/github_api.py [hunk 2]
```diff
@@ -729,22 +730,23 @@ def _not_duplicate(pr_messages, new_message):
...
         """
         Sends a message that this PRs commits have been deployed to the staging environment
 
         Args:
-            pr_number (int): The number of the pull request
+            pr (github.PullRequest.PullRequest or int): The pull request (object or number) to message
```
Comment (jibsheet) [inline tubular/github_api.py]: Should this be the same var name as above (pull_request) ?

File: tubular/github_api.py [hunk 3]
```diff
@@ -755,45 +757,45 @@ def message_pr_deployed_stage(self, pr_number, deploy_date=None, force_message=F
...
         )
 
-    def message_pr_deployed_prod(self, pr_number, force_message=False):
+    def message_pr_deployed_prod(self, pr, force_message=False):
         """
         sends a message that this PRs commits have been deployed to the production environment
 
         Args:
-            pr_number (int): The number of the pull request
+            pr (github.PullRequest.PullRequest or int): The pull request (object or number) to message
```
Comment (jibsheet) [inline tubular/github_api.py]: same question?

File: tubular/github_api.py [hunk 4]
```diff
@@ -755,45 +757,45 @@ def message_pr_deployed_stage(self, pr_number, deploy_date=None, force_message=F
...
         )
 
-    def message_pr_release_canceled(self, pr_number, force_message=False):
+    def message_pr_release_canceled(self, pr, force_message=False):
         """
         Sends a message that this PRs commits have not made it to production as the release was canceled
 
         Args:
-            pr_number (int): The number of the pull request
+            pr (github.PullRequest.PullRequest or int): The pull request (object or number) to message
```
Comment (jibsheet) [inline tubular/github_api.py]: same question?

**Review by jibsheet — APPROVED** [review]
var naming question, but otherwise looks fine

**Pull request merged**